src/datachain/lib/file.py

```diff
@@ -206,6 +207,19 @@ def upload(
         file._set_stream(catalog)
         return file
 
+    @classmethod
+    def from_local_path(cls, path: str, catalog: Optional["Catalog"] = None) -> "File":
+        if catalog is None:
+            from datachain.catalog.loader import get_catalog
+
+            catalog = get_catalog()
+
+        parent, name = posixpath.split(path)
+        client = catalog.get_client(parent)
```

Comment:
[C] The top of this method is the same as `upload` could have a `get_client_from_path` helper